### PR TITLE
custom parameters

### PR DESCRIPTION
**Pull request by MegaByte875**

### CODE REVIEW

File: doc/user/custom_config.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    image: vesoft/nebula-graphd
+    version: v2.0.0
+    storageClaim:
+      resources:
+        requests:
+          storage: 2Gi
+      storageClassName: gp2
+    config:
+      "enable_authorize": "true"
+      "auth_type": "password"
```
Comment (veezhang) [inline doc/user/custom_config.md:35]: How to get the password? Or how to set it? 
And is there has some ref link for password.

File: pkg/util/config/config.go [hunk 2]
```diff
@@ -17,27 +17,54 @@ limitations under the License.
...
+				delete(custom, param)
+			}
+			b.WriteString(fmt.Sprintf("--%s=%s\n", param, value))
+		}
 	}
-	_, err = b.WriteString(config)
-	if err != nil {
-		return data
+
+	if len(custom) > 0 {
```
Comment (veezhang) [inline pkg/util/config/config.go:59]: Should add `if len(custom) ==0 { return data } `  at the front of this func?

File: pkg/util/config/config_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+########## Custom ##########
+--enable_optimizer=true
+`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := AppendCustomConfig(tt.args.data, tt.args.custom); got != tt.want {
+				fmt.Println(got)
```
Comment (veezhang) [inline pkg/util/config/config_test.go:94]: remove debug code.

File: pkg/util/config/config.go [hunk 4]
```diff
@@ -17,27 +17,54 @@ limitations under the License.
...
-	if err != nil {
-		return data
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if line == "" || strings.HasPrefix(line, "#") {
+			b.WriteString(fmt.Sprintf("%s\n", line))
+		}
+
+		if strings.HasPrefix(line, "--") {
```
Comment (veezhang) [inline pkg/util/config/config.go:43]:
```shell
if !strings.HasPrefix(line, "--") {
    b.WriteString(fmt.Sprintf("%s\n", line))
}

// do modify configs.
```

File: pkg/util/config/config.go [hunk 5]
```diff
@@ -17,27 +17,54 @@ limitations under the License.
...
+
+	if len(custom) > 0 {
+		b.WriteString("\n########## Custom ##########\n")
+	}
+	for k, v := range custom {
+		b.WriteString(fmt.Sprintf("--%s=%s\n", k, v))
+	}
+
+	if err := scanner.Err(); err != nil {
+		klog.Errorf("reading input failed:", err)
```
Comment (veezhang) [inline pkg/util/config/config.go:67]: Maybe this code move after the for loop, for better locality.

File: pkg/util/config/config.go (hunk 2 above)
Comment (veezhang) [reply pkg/util/config/config.go:59]: And there 
```go
if len(custom) > 0 {
    return b.String()
}

// append the custom configure.
...
````

File: pkg/util/config/config_test.go [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+func TestAppendCustomConfig(t *testing.T) {
+	type args struct {
+		data   string
+		custom map[string]string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
```
Comment (veezhang) [inline pkg/util/config/config_test.go:27]: Generally, the anonymous structure is used in test cases.
```shell
tests := []struct {
    data   string
    custom map[string]string
}{{
    data: "...",
    ...
},{
}}
```

**Comment by veezhang** [comment]
@MegaByte875 Configuration parameters should be limited for not all of configs are supported.

File: doc/user/custom_config.md (hunk 1 above)
Comment (MegaByte875) [reply doc/user/custom_config.md:35]: This is an example that  show user how to use config

**Review by veezhang — APPROVED** [review]
LGTM

**Pull request merged**